### ISSUE TITLE
S2333 - False positive on generic base classes

### ISSUE THREAD

**Issue by njannink** [SonarSource/sonarlint-vs#523]
Hello I get a false positive on generic abstract class with virtual methods.

Example code:

```
    internal abstract class BaseClass<T>
    {
        public virtual string Process(string input) => input; //Compliant
    }

    internal class SubClass : BaseClass<string>
    {
        public override string Process(string input)
        {
            return "Test";
        }
    }
```

**Comment by dbolkensteyn** [comment]
LGTM
